### ISSUE TITLE
new server of ASKY is now online.

### ISSUE THREAD

**Issue by redbain** [browserquest/BrowserQuest#152]
http://xe17.conory.com/bbo/index.html

We opened a new server of browserquest-ASKY,
When trying to log-in for the first time, it takes a while to load..
Is this because of the size of the images?
Any suggestions to solve this problem?

**Comment by Flavius256** [comment]
@phinlolex Yes, I have managed to implement the ban system from ASKY into my personal extended version of BrowserQuest, you just need to be admin and tweak around with the commands.

**Comment by phinlolex** [comment]
@Flavius256 Hello,
Thank you very much for your response ! :smile: 
https://github.com/browserquest/BrowserQuest-Asky <- this is your personal version?

**Comment by Flavius256** [comment]
Mine is this repository with various fixes I will release an older version soon but I can assure you the ban system works except for the "player.connection.close();" in checkBan in the databaseHandler class. Aside from that I found a work around that by placing the player in a dark zone to prevent movement and mute them for the duration of the ban. I will see if there is a way to properly close the connection.

**Comment by phinlolex** [comment]
@Flavius256 Hello,

When you say "tweak around with the commands",i must write a command in the ubuntu terminal, or I have to change a javascript code?
(I'm a beginner in the computer field.)

**Comment by Flavius256** [comment]
@phinlolex in player.js (server-side) there is a section that handles commands that are written (search for ASKY) and it should get you there.

**Comment by phinlolex** [comment]
@Flavius256 I am in the player.js file (server side), I do not see the section that manages orders. Can you help me please?

**Comment by Flavius256** [comment]
@phinlolex Alright so firstly we will need a javascript implementation for the use of .contains, .startsWith, .endsWith.
Before starting that you firstly need to open up Player.js (server side) and search for:

log.info("CHAT: " + self.name + ": " + msg);

Below (one or two lines down) you should see something like:

if(msg && msg !== "") {

and underneath that there is a Substring declaring the length of the message like so:

 msg = msg.substr(0, 60);

--------------------------------------

Now we need to declare the startsWith method for a better way of reading commands compared to Asky's.

So right under the msg = msg.substr(0, 60); add the following:

 if ( typeof String.prototype.startsWith !== 'function' ) {
                                String.prototype.startsWith = function( str ) {
                                    return str.length > 0 && this.substring( 0, str.length ) === str;
                        };
                    };

                    if ( typeof String.prototype.endsWith !== 'function' ) {
                        String.prototype.endsWith = function( str ) {
                            return str.length > 0 && this.substring( this.length - str.length, this.length ) === str;
                        };
                    };

Alright, this will allow us to read commands. Firstly we will add a ban command, note that it is sort of buggy and will cause some errors the key to fixing this is in redis.js in the checkBan method, I made it so the player is transported to an empty area as long as he is banned until I find a fix.

Add this under the last code we've added:
var targetPlayer = self.server.getPlayerByName(msg.split(' ')[1]);  


Now for the ban method, we can handle this using a switch or if/else statement depending on what you prefer. Now let's handle the ban method. Under the targetPlayer variable add:

if(msg.startsWith("/ban ")) {
                        
                        var banPlayer = self.server.getPlayerByName(msg.split(' ')[2]);
                        var days = (msg.split(' ')[1])*1;
                        if(banPlayer) {
                            databaseHandler.setPlayerPosition(banPlayer, 154, 4);
                            databaseHandler.banPlayer(self, banPlayer, days);

                       }

}

After restarting or starting the server you can go online and attempt to ban yourself using the following syntax: /ban 1 yourname

If you want to change the syntax you can change:
                        var banPlayer = self.server.getPlayerByName(msg.split(' ')[2]);
                        var days = (msg.split(' ')[1])*1;

to:

                        var banPlayer = self.server.getPlayerByName(msg.split(' ')[1]);
                        var days = (msg.split(' ')[2])*1;

This inverts the last two statements order so the syntax will be: /ban yourname 1

Note: I think that is everything for implementing it, don't forget to check the checkBan method in redis.js as the problem is always when the player tries to log in the second time, the error is exactly at the following line:

player.connection.close("blablabla");

Also, if you want to change the amount of time the player remains on before being kicked off the server do so in redis.js in the method banPlayer on the line that starts with:

setTimeout( function(){ banPlayer.connection.close

If I remember correctly at the end of the line there is a 30000 number (in milliseconds) that indicates the player will be kicked off in 30 seconds, you can change that to whatever you please. I hope this was helpful, I really wish I had other people helping me with this game.

**Comment by phinlolex** [comment]
@Flavius256 Thank you very much to have answered me!

I modified stage by stage the file player.js (server-side). 

What gives me this :

                log.info("CHAT: " + self.name + ": " + msg);

                // Sanitized messages may become empty. No need to broadcast empty chat messages.
                if(msg && msg !== "") {
                    msg = msg.substr(0, 60); 
					if ( typeof String.prototype.startsWith !== 'function' ) { String.prototype.startsWith = function( str ) { return str.length > 0 && this.substring( 0, str.length ) === str; }; }; if ( typeof String.prototype.endsWith !== 'function' ) { String.prototype.endsWith = function( str ) { return str.length > 0 && this.substring( this.length - str.length, this.length ) === str; }; };
                    var targetPlayer = self.server.getPlayerByName(msg.split(' ')[1]); 
					    if(msg.startsWith("/ban ")) {

                     var banPlayer = self.server.getPlayerByName(msg.split(' ')[1]);
                    var days = (msg.split(' ')[2])*1;
                    if(banPlayer) {
                        databaseHandler.setPlayerPosition(banPlayer, 154, 4);
                        databaseHandler.banPlayer(self, banPlayer, days);

                   }
                }


But, when I open the server (node server/js/main.js) and when I write the command "/ban phinolex 1" there is nothing going.

![github](https://cloud.githubusercontent.com/assets/1743597/5908758/a206c86c-a5aa-11e4-9dee-97c4c2c02da4.png)


What should I do?

**Comment by Flavius256** [comment]
@phinlolex 

Just try to use the original banPlayer and days variables and see if it works the other way around

                       var banPlayer = self.server.getPlayerByName(msg.split(' ')[2]);
                        var days = (msg.split(' ')[1])*1
Also before going any further, make sure you your name capitalized accordingly

**Comment by phinlolex** [comment]
@Flavius256 Hello,

I have to write the command "/ban player 1" in the ubuntu terminal?

**Comment by Flavius256** [comment]
@phinlolex No, the commands are using an admin account. You also have to make yourself admin, I forgot that. In redis.js in the loadPlayer method add client.sadd("yourusername");

**Comment by phinlolex** [comment]
I don't understand.
How can I do to be admin? What should I edit in the file "redis.js" for make me admin?

**Comment by phinlolex** [comment]
@Flavius256
 I don't understand.
How can I do to be admin? What should I edit in the file "redis.js" for make me admin?

**Comment by phinolex** [comment]
@Flavius256  Hello. :)
I followed step by step what you have told me.

So in the file server-> player.js, I add:

```
   if ( typeof String.prototype.startsWith !== 'function' ) {
                        String.prototype.startsWith = function( str ) {
                            return str.length > 0 && this.substring( 0, str.length ) === str;
                };
            };

            if ( typeof String.prototype.endsWith !== 'function' ) {
                String.prototype.endsWith = function( str ) {
                    return str.length > 0 && this.substring( this.length - str.length, this.length ) === str;
                };
            };
            var targetPlayer = self.server.getPlayerByName(msg.split(' ')[1]);
             if(msg.startsWith("/ban ")) {

                var banPlayer = self.server.getPlayerByName(msg.split(' ')[2]);
                var days = (msg.split(' ')[1])*1;
                if(banPlayer) {
                    databaseHandler.setPlayerPosition(banPlayer, 154, 4);
                    databaseHandler.banPlayer(self, banPlayer, days);

               }
            }
```

If I place this code in the player.js file, I have:

http://phinolex.free.fr/player.js

You told me , I have to be a administrator. So I add as you said "client.sadd (" phinolex ");" in redis.js file:

http://phinolex.free.fr/redis.js

I saved the changes I made. I restarted the server.
I log under the username "phinolex" like this:

![phinolex-browserquest](https://cloud.githubusercontent.com/assets/1743597/6314669/7aed1496-b9e8-11e4-8f7b-4afa2003e74e.png)

I created a second account "test".

![test-browserquest](https://cloud.githubusercontent.com/assets/1743597/6314690/4765babe-b9e9-11e4-92e0-76713c631c32.png)

When I write the command "/ ban 1 test" there is nothing. The player "test" is always present on the game. :/
![ban-test](https://cloud.githubusercontent.com/assets/1743597/6314748/02d92a82-b9eb-11e4-8cbd-26892e35e5ac.png)

What should I do?
I made an error in the javascript code?

Help me please. :'(

**Comment by Radelas** [comment]
I test the command /ban 1 "pseudo" and i've got an error : 
![erreur](https://cloud.githubusercontent.com/assets/11009805/7006594/bd1e7fa0-dc82-11e4-8c9a-b8e624c16cd4.png)

What is the problem ?
